Move Transport UDF examples to a separate gradle project

shardulm94 opened the pull request:
Moved Transport UDF example modules into its own gradle project with independent `build.gradle` and `settings.gradle` files. This decouples the build of the examples from the core project thus allowing us to treat them as independent projects. With composite builds we can still reference the core dependencies built in the repo rather than referring to published binaries. This allows us to test any changes to core dependencies by applying them on the examples before publishing the core binaries.

The `build.gradle` file is very barebones right now, you can run `build` on it. These artifacts will no longer be published and that is acceptable since they are not meant to be consumed but rather meant to serve as documentation. If we need to be publish artifacts in the future, we can do so by configuring a publishing mechanism similar to the core project.

I removed the old tests from the example modules; they relied on the non-default configuration `tests` which is not supported in composite builds. These tests are already replicated using the new unified testing framework which are intact (note: `avro` tests are not replicated).

I also updated the `travis.yml` file to build the examples project as part of the build pipeline so that we can catch failures in the examples due to changes in the core.

rdsr commented on transportable-udfs-examples/settings.gradle:
```diff
@@ -0,0 +1,20 @@
+
+rootProject.name = 'transportable-udfs-examples'
+
+includeBuild('../')
```
Will our IDE understand this? Have we tried importing this examples project in Intellij and see it resolves the transport dependencies correctly?

shardulm94 replied:
IDE (atleast IntelliJ Idea) will understand the `includeBuild` statement correctly. It will build the included build before building the examples project. I have verified this. If we open the examples project in the IDE, it will also automatically import the core project and we are able to work on both projects simultaneously (atleast IntelliJ Idea).
![image](https://user-images.githubusercontent.com/6961317/54716673-7cb1ce00-4b13-11e9-8df0-d1bbb8fa8651.png)

rdsr replied:
Thanks for checking @shardulm94

shardulm94 commented:
A few changes made in the above commits
- Rebased this PR on the current master
- Added a gradle wrapper for the `transportable-udfs-examples` project
- Enabled license and style checks for the `transportable-udfs-examples` project
- Fixed backward incompatibility of checkstyle rules for an updated checkstyle version in Gradle 5.0
- Use HTTPS instead of HTTP for the conjars repo (Same as #14 but for the `transportable-udfs-examples` project)
- Updated `.travis.yml` to short circuit on build failures and not go to the release step.
